File: largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        string res ;
        int l = num.length();
        for(int i = 0 ; i < l-2 ; i++){
            if(num[i]==num[i+1] && num[i+1]==num[i+2]){
                string ans;
                ans.push_back(num[i]);
                ans.push_back(num[i+1]);
                ans.push_back(num[i+2]);
                res = max(ans,res);
            }
        }
        return res;
    }
};

File: largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        res = []
        if len(num)<3:
            return ""

        for i in range(len(num)-2):
            if num[i]==num[i+1] and num[i+1]==num[i+2]:
                res.append(num[i:i+3])
        res = sorted(res)
        if len(res):
            return res[-1]
        return ""
        